### PR TITLE
Phase 12-D: parallel CUDA rowwise fallback attempt

### PR DESCRIPTION
**Pull request by ericflo**
## Summary\n- Attempts to recover PR #1004 c=8 throughput by dispatching safe CUDA rowwise fallback rows concurrently.\n- Adds a replay-ready gate so row workers do not concurrently CUDA-graph-capture on Candle's shared CUDA stream.\n- Keeps PR #1005's cuBLAS+reduce argmax path closed and does not materialize batched CUDA LM-head logits.\n\n## Validation\n- PASS: cargo fmt --check\n- PASS: KILN_CUDA_ARCHS=86 cargo build --release --features cuda --bin kiln\n- PASS: KILN_CUDA_ARCHS=86 cargo nextest run --release -p kiln-model cuda_graph --features cuda\n- FAIL/known unrelated: KILN_CUDA_ARCHS=86 cargo nextest run --release -p kiln-model --features cuda hit existing CUDA kernel parity failures in test_causal_conv1d_update_matches_fallback and test_gdn_chunk_body_matches_fallback.\n- PASS: W4A16 dp1-vs-dp8 parity oracle, 0/256 selected-token mismatches.\n- FAIL: short Shape A c=1 smoke 43.81 tok/s (<49.30 floor), Shape A c=8 smoke 50.87 tok/s (<62.45 baseline), reliability clean.\n\nReport: b2://clouderic/diagnostics/kiln-phase12-d-rowwise-recovery-validation-2026-05-09/REPORT.md\n\n## Verdict\nDo not merge. Thread-level overlap is not enough to recover c=8, and the first ungated attempt showed concurrent CUDA graph capture invalidation. This PR is intentionally draft and should be closed as a measured regression.

### CODE REVIEW

**Comment by ericflo** [comment]
Closing intentionally: validation failed throughput gates. Report: b2://clouderic/diagnostics/kiln-phase12-d-rowwise-recovery-validation-2026-05-09/REPORT.md\n\nKey results: Shape A c=1 smoke 43.81 tok/s (<49.30), Shape A c=8 smoke 50.87 tok/s (<62.45), reliability clean (0 OOM/illegal-address, 0 HTTP 5xx/408, all counted rows length). The ungated parallel version invalidated CUDA graph capture; the replay-ready-gated version is safe but slower. Do not merge.
